src/test/java/org/broadinstitute/hellbender/engine/cache/LocatableCacheUnitTest.java

```diff
@@ -0,0 +1,431 @@
+    public void testFeatureCacheFeatureRetrieval( final LocatableCache<ArtificialTestFeature> cache, final List<Pair<Integer, List<ArtificialTestFeature>>> retrievalOperations ) {
+        for ( Pair<Integer, List<ArtificialTestFeature>> retrievalOperation: retrievalOperations ) {
+            final int stopPosition = retrievalOperation.getLeft();
+            final List<ArtificialTestFeature> expectedFeatures = retrievalOperation.getRight();
+
+            final List<ArtificialTestFeature> actualFeatures = cache.getCachedLocatables(new SimpleInterval(cache.getContig(), cache.getCacheStart(), stopPosition));
+            Assert.assertEquals(actualFeatures, expectedFeatures, "Wrong Features returned in retrieval operation with stop position " + stopPosition);
+        }
+    }
+
+    /**
+     * Test caching a region with no Features. This should work (we should avoid going to disk
+     * to look for new records when querying within such a region).
+     */
+    @Test
+    public void testHandleCachingOfEmptyRegion() {
+        List<ArtificialTestFeature> emptyRegion = new ArrayList<>();
+        // initialize cache with 0 lookahead bases
+        LocatableCache<ArtificialTestFeature> cache = new LocatableCache<>(
+                "test",
+                new DrivingFeatureInputCacheStrategy<ArtificialTestFeature>(
+                        0, (Locatable newInterval) -> emptyRegion.iterator()));
+
+        // prime the cache with an empty iterator
+        cache.queryAndPrefetch(new SimpleInterval("1", 1, 100));
+
+        Assert.assertTrue(cache.isEmpty(), "Cache should be empty");
+        Assert.assertTrue(cache.cacheHit(new SimpleInterval("1", 1, 100)), "Unexpected cache miss");
+        Assert.assertTrue(cache.cacheHit(new SimpleInterval("1", 2, 99)), "Unexpected cache miss");
+
+        // query the empty region to force a (no-op) trimCache, after which we still expect a cacheHit
+        Assert.assertEquals(
+                cache.queryAndPrefetch(
+                    new SimpleInterval(cache.getContig(), 2, cache.getCacheEnd())),
+                    emptyRegion);
+        Assert.assertTrue(cache.cacheHit(new SimpleInterval("1", 2, 100)), "Unexpected cache miss");
+
+        // now query again and make sure we don't do another query
+        Assert.assertEquals(
+                cache.queryAndPrefetch(
+                        new SimpleInterval(cache.getContig(), cache.getCacheStart(), 100)),
+                emptyRegion,
+                "Should get back empty List for empty region");
+    }
+
+    /***************************************************************************************************
+     * End of LocatableCache tests using DrivingFeatureInputCacheStrategy to test basic cache operations
+     ***************************************************************************************************/
+
+    /********************************************************
+     * LocatableCache tests using SideReadInputCacheStrategy
```

Comment:
This test is really testing `SideReadInputCacheStrategy`,so maybe it should be moved to a SideReadInputCacheStrategyUnitTest class.